include/kodlab_mjbots_sdk/mjbots_control_loop.h

```diff
@@ -161,15 +161,18 @@ template<class log_type, class input_type, class robot_type>
 MjbotsControlLoop<log_type, input_type, robot_type>::MjbotsControlLoop(std::vector<std::shared_ptr<kodlab::mjbots::JointMoteus>> joint_ptrs, const ControlLoopOptions &options)
   : AbstractRealtimeObject(options.realtime_params.main_rtp, options.realtime_params.can_cpu),
     lcm_sub_(options.realtime_params.lcm_rtp, options.realtime_params.lcm_cpu, options.input_channel_name) 
-{
-  robot_ = std::make_shared<robot_type>(  joint_ptrs,
-                                          options.soft_start_duration,
-                                          options.max_torque);
-  
+{  
   mjbots_interface_ = std::make_shared<kodlab::mjbots::MjbotsHardwareInterface>(joint_ptrs,
                                                                                 options.realtime_params,
                                                                                 options.imu_mounting_deg,
-                                                                                options.attitude_rate_hz);
+                                                                                options.attitude_rate_hz, 
+                                                                                options.imu_world_offset_deg);
+
+  robot_ = std::make_shared<robot_type>(  joint_ptrs,
+                                          options.soft_start_duration,
+                                          options.max_torque,
+                                          mjbots_interface_->GetIMUDataSharedPtr());
```

Comment:
When using joints-based-constructor for control loop, make mjbots_interface_ first and then make robot with imu_data_ptr as parameter.